bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -265,6 +277,36 @@ object BagsManager {
   // This is our mutable state.  It is only read/written by the Swing thread.
   private var stateMap  = ==>>.empty[ProgKey, ObsKey ==>> BagsState]
 
+  def statusLookup(p: SPNodeKey, o: SPNodeKey): Option[BagsStatus] =
+    for {
+      obsMap <- stateMap.lookup(ProgKey(p))
+      s      <- obsMap.lookup(ObsKey(o))
+      st     <- s.status
+    } yield st
+
+  // Listeners for changes to the Bags Status.
+  // Management must be done on the Swing EDT.
+  private var listeners: List[BagsStatusListener] = Nil
+
+  def addBagsStatusListener(l: BagsStatusListener): Unit = {
```

Comment:
Could it make sense that instead of checking if we are on the EDT we just run it on `Swing.onEDT`?